src/integrationTest/java/uk/gov/hmcts/reform/fprl/DocumentManagementServiceImplIntegrationTest.java

```diff
@@ -0,0 +1,183 @@
+            .andReturn();
+
+        assertEquals(response.getStatusCode(), HttpStatus.SC_OK);
+    }
+
+    @Test
+    public void checkHealthStatusOfCCDCaseDocumentApiIsUp_thenReturn200Status() throws Exception {
+
+        Response response = SerenityRest.given()
+            .when()
+            .get(constructHealthUrl(CCDDOCUMENTAPI))
+            .andReturn();
+
+        assertEquals(response.getStatusCode(), HttpStatus.SC_OK);
+    }
+
+    @Test
+    public void givenInvalidTemplate_whenRequestMade_thenReturn400Response() throws Exception {
+
+        String requestBody = ResourceLoader.loadJson(INVALID_TEMPLATE_DATA_JSON);
+
+        Response response = callPrlDocumentGenerator(requestBody);
+
+        assertEquals(
+            400,
+            response.getStatusCode()
+        );
+    }
+
+    @Test
+    public void givenTemplateAndJson_whenDocumentGenerated_thenCorrectFormat() throws Exception {
+
+        String requestBody = ResourceLoader.loadJson(VALID_INPUT_JSON);
+        Response response = callPrlDocumentGenerator(requestBody);
+        GeneratedDocumentInfo generatedDocumentInfo = response.as(GeneratedDocumentInfo.class);
+
+        String mimeType = generatedDocumentInfo.getMimeType();
+        String expectedMimeType = "application/pdf";
+
+        assertEquals(mimeType, expectedMimeType);
+    }
+
+    @Test
+    public void givenValidAccessToken_whenAccessingDocumentPath_thenAccessProvided() throws Exception {
+
+        String requestBody = ResourceLoader.loadJson(VALID_INPUT_JSON);
+        Response response = callPrlDocumentGenerator(requestBody);
+        GeneratedDocumentInfo generatedDocumentInfo = response.as(GeneratedDocumentInfo.class);
+
+        String[] dmStoreUrlParts = generatedDocumentInfo.getUrl().split("/");
+
+        String documentId = dmStoreUrlParts[dmStoreUrlParts.length - 1];
+
+        String url = "https://gateway-ccd.aat.platform.hmcts.net/documents/" + documentId; //todo url
+
+        String bearerToken = getAuthorizationToken();
+
+        HttpGet request = new HttpGet(url);
+
+        request.setHeader(HttpHeaders.AUTHORIZATION, bearerToken);
+        HttpResponse httpResponse = HttpClientBuilder.create().build().execute( request );
+
+        assertEquals(
+            httpResponse.getStatusLine().getStatusCode(),
+            HttpStatus.SC_OK);
+    }
+
+    @Test
+    public void givenInvalidAccessToken_whenAccessingDocumentPath_thenNoAccessProvided() throws Exception {
+
+        String requestBody = ResourceLoader.loadJson(VALID_INPUT_JSON);
+        Response response = callPrlDocumentGenerator(requestBody);
+        GeneratedDocumentInfo generatedDocumentInfo = response.as(GeneratedDocumentInfo.class);
+
+        String[] dmStoreUrlParts = generatedDocumentInfo.getUrl().split("/");
+
+        String documentId = dmStoreUrlParts[dmStoreUrlParts.length - 1];
+
+        String url = "https://gateway-ccd.aat.platform.hmcts.net/documents/" + documentId; //todo url in properties
```

Comment:
move url to property file and remove todo comment

Reply:
Url moved to Properties